testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -822,6 +822,42 @@ public void check3pCookiesMessageCallbackTest() {
                 .init(defaultArguments(), this::assertInitNotAuth);
     }
 
+    // In case of incorrect/unavailable realm provided in KeycloakConfig,
+    // JavaScript Adapter init() should fail-fast and reject Promise with KeycloakError.
+    @Test
+    public void checkInitWithInvalidRealm() {
+
+        JSObjectBuilder keycloakConfig = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", "invalid-realm-name")
+                .add("clientId", CLIENT_ID);
+
+        JSObjectBuilder initOptions = defaultArguments();
+
+        testExecutor
+                .configure(keycloakConfig)
+                .init(initOptions, this::assertErrorResponse);
+
+    }
+
+    // In case of unavailable Authorization Server due to network or other kind of problems,
+    // JavaScript Adapter init() should fail-fast and reject Promise with KeycloakError.
+    @Test
+    public void checkInitWithUnavailableAuthServer() {
+
+        JSObjectBuilder keycloakConfig = JSObjectBuilder.create()
+                .add("url", "https://localhost:12345/auth")
```

Comment:
This is not very nice :-(. But I can't think of any better solution. If nobody has a better idea we can leave this as is for now, and investigate if there are any failures in the future.

Reply:
This is test environment/infrastructure task to provide **reliable** TCP service that can simulate TCP SYN-RST / ICMP Destination Unreachable / High latency and other network failures.